### DISCUSSION TITLE
[질문] 이번 지하철 경로 조회 미션에서 다들 상태코드를 어떻게 사용하셨나요?

### DISCUSSION THREAD

**Discussion by jurlring** [woowacourse-study/2022-http-web-basic-for-all-developer#31] [Q&A]
**주디**
1. IllegalStateException => 404 NOT FOUND
2. IllegalArgumentException => 400 BAD REQUEST
3. MethodArgumentNotValidException => 400 BAD REQUEST
4. Exception => 500 INTERNAL SERVER ERROR

**Comment by RunaNam** [comment]
- NotFoundException (커스텀 익셉션) -> 404 Not Found
- IllegalArgumentException -> 400 Bad Request
- Exception -> 500 Internal Server Error

**Comment by kth990303** [comment]
웬만해선 400 Bad Request
런타임에러 등 요청이 잘못된 에러가 아닌 경우 500 Internal Server Error
* * *
### 잘못된 id로 접근했을 때 400 vs 404
저는 400을 선택했습니다. 클라이언트 입장에서 404를 봤을 때 해당 리소스 자체가 존재하지 않는 것으로 판단할 여지가 있다고 생각했기 때문이에요. 잘못된 id 요청을 한 것이라 판단하기도 해서 400을 이용했습니다.

하지만 같은 상황에서의 404도 충분히 좋은 상태코드 처리라 생각합니다.

**Comment by hyeonic** [comment]
NotFoundException (커스텀 익셉션) -> 404 Not Found
IllegalArgumentException -> 400 Bad Request
Exception -> 500 Internal Server Error

루나랑 동일하네요! 

@kth990303 
`id와 동일한 타입 && 정상 범위의 id && 존재하지 않을 때` -> 404
`id와 다른 타입 || 비정상적인 범위의 id` -> 400
위와 같은 기준으로 두 가지 상태 코드를 적절히 활용할 것 같네요!
